examples/applications/cloud-app/src/Program.cs

```diff
@@ -2,109 +2,482 @@
+                        Debug.Assert(valueText is not null);
+
+                        Console.WriteLine(TryParseValue(valueText, out var kind, out var value)
+                                          ? $"{kind}: {value}"
+                                          : "ERROR");
+                        break;
+                    }
+                    case { CmdInspect: true, ArgNamespace: var ns, ArgQuery: var query }:
+                    {
+                        Debug.Assert(ns is not null);
+
+                        request = FulfillRequest(ns, fi => fi.Inspect = new() { Query = query });
+                        break;
+                    }
+                    case { CmdRead: true, ArgNamespace: var ns, ArgKey: var key }:
+                    {
+                        Debug.Assert(ns is not null);
+
+                        request = FulfillRequest(ns, fi => fi.Read = new() { Key = key });
+                        break;
+                    }
+                    case { CmdWrite: true, ArgNamespace: var ns, ArgKey: var key, ArgValue: var valueText }:
+                    {
+                        Debug.Assert(ns is not null);
+                        Debug.Assert(valueText is not null);
+
+                        if (TryParseValue(valueText, out _, out var value))
+                            request = FulfillRequest(ns, fi => fi.Write = new() { Key = key, Value = value });
+                        else
+                            Console.Error.WriteLine($"Invalid value: {valueText}");
+                        break;
+                    }
+                    case { CmdInvoke: true, ArgNamespace: var ns, ArgCommand: var cmd, ArgArg: var cmdArgs }:
+                    {
+                        Debug.Assert(ns is not null);
+                        Debug.Assert(cmdArgs is not null);
+
+                        var argValuePairs =
+                            cmdArgs.Select(arg => TryParseValue(arg, out _, out var v) ? (Arg: arg, Value: v) : (arg, null))
+                                   .ToList();
+
+                        if (argValuePairs.All(av => av is (_, not null)))
+                        {
+                            request = FulfillRequest(ns, fi =>
+                            {
+                                var invokeIntent = new InvokeIntent { Command = cmd };
+                                invokeIntent.Args.AddRange(from av in argValuePairs select av.Value);
+                                fi.Invoke = invokeIntent;
+                            });
+                        }
+                        else
+                        {
+                            Console.Error.WriteLine($"Invalid value: {argValuePairs.First(v => v.Value is null).Arg}");
+                        }
+                        break;
+                    }
+                    case { CmdSubscribe: true, ArgNamespace: var ns, ArgSource: var sources }:
+                    {
+                        Debug.Assert(ns is not null);
```

Comment:
Should we consider creating the events file if it does not yet exist if we subscribe? When I tried tail the file when no event was published, I had to follow a specific order (publish inside the car, then tail) to get this done. It might easier for the developer if we `touch` the file, even if we received no events so far.

Reply:
Good idea and done with cbe727f6f7142f715a88887a084a5757a4280e31.